src/audio/node.rs

```diff
@@ -41,20 +41,26 @@ impl BlockInfo {
     }
 }
 
+/// This trait represents the common features of all audio nodes.
 pub trait AudioNodeEngine: Send {
-    // XXX Create an AudioBuffer abstraction
-    fn process(
-        &mut self,
-        inputs: Chunk,
-        info: &BlockInfo,
-    ) -> Chunk;
+    fn process(&mut self, inputs: Chunk, info: &BlockInfo) -> Chunk;
 
-    fn message(&mut self, _: AudioNodeMessage, _sample_rate: f32) {
-
-    }
+    fn message(&mut self, _: AudioNodeMessage, _sample_rate: f32) {}
 }
 
-
 pub enum AudioNodeMessage {
-    SetAudioParamEvent(UserAutomationEvent)
+    OscillatorNode(OscillatorNodeMessage),
+    GainNode(GainNodeMessage),
```

Comment:
I think we should be able to send all kinds of messages to all nodes, and nodes filter what they can actually handle.

Also, this reverts the AudioParam changes.

So you have SetAudioParam, Start, and Stop.